code/controllers/subsystem/polling.dm

```diff
@@ -0,0 +1,216 @@
+	var/list/candidates = list()
+	if(!(GLOB.ghost_role_flags & GHOSTROLE_STATION_SENTIENCE))
+		return candidates
+
+	for(var/mob/dead/observer/ghost_player in GLOB.player_list)
+		candidates += ghost_player
+
+	return poll_candidates(question, role, check_jobban, poll_time, ignore_category, flashwindow, candidates, pic_source, role_name_text)
+
+/datum/controller/subsystem/polling/proc/poll_ghost_candidates_for_mob(question, role, check_jobban, poll_time = 30 SECONDS, mob/target_mob, ignore_category = null, flashwindow = TRUE, pic_source, role_name_text)
+	var/static/list/mob/currently_polling_mobs = list()
+
+	if(currently_polling_mobs.Find(target_mob))
+		return list()
+
+	currently_polling_mobs += target_mob
+
+	var/list/possible_candidates = poll_ghost_candidates(question, role, check_jobban, poll_time, ignore_category, flashwindow, pic_source, role_name_text)
+
+	currently_polling_mobs -= target_mob
+	if(!target_mob || QDELETED(target_mob) || !target_mob.loc)
+		return list()
+
+	return possible_candidates
+
+/datum/controller/subsystem/polling/proc/poll_ghost_candidates_for_mobs(question, role, check_jobban, poll_time = 30 SECONDS, list/mobs, ignore_category = null, pic_source, role_name_text)
+	var/list/candidate_list = poll_ghost_candidates(question, role, check_jobban, poll_time, ignore_category, pic_source, role_name_text)
+
+	for(var/mob/potential_mob as anything in mobs)
+		if(QDELETED(potential_mob) || !potential_mob.loc)
+			mobs -= potential_mob
+
+	if(!length(mobs))
+		return list()
+
+	return candidate_list
+
+/datum/controller/subsystem/polling/proc/is_eligible(mob/potential_candidate, role, check_jobban, the_ignore_category)
+	if(!potential_candidate.key || !potential_candidate.client)
+		return FALSE
+	if(the_ignore_category)
+		if(potential_candidate.ckey in GLOB.poll_ignore[the_ignore_category])
+			return FALSE
+	if(role)
+		if(!(role in potential_candidate.client.prefs.be_special))
+			return FALSE
+		var/required_time = GLOB.special_roles[role] || 0
+		if(potential_candidate.client && potential_candidate.client.get_remaining_days(required_time) > 0)
+			return FALSE
+
+	if(check_jobban)
+		if(is_banned_from(potential_candidate.ckey, list(check_jobban, ROLE_SYNDICATE)))
+			return FALSE
+
+	return TRUE
+
+/datum/controller/subsystem/polling/proc/polling_finished(datum/candidate_poll/finishing_poll)
+	// Trim players who aren't eligible anymore
+	var/length_pre_trim = length(finishing_poll.signed_up)
+	finishing_poll.trim_candidates()
+	log_game("Candidate poll [finishing_poll.role ? "for [finishing_poll.role]" : "\"[finishing_poll.question]\""] finished. [length_pre_trim] players signed up, [length(finishing_poll.signed_up)] after trimming")
+
+	finishing_poll.finished = TRUE
+	currently_polling -= finishing_poll
+
+	// Determine which is the next poll closest the completion or "disable" firing if there's none
+	if(!length(currently_polling))
```

Comment:
Nit:

```suggestion
	if(currently_polling.len > 0)
```

- `length` obscures errors with null
- Not great to use `0` as falsy